compiler/src/dotty/tools/dotc/transform/InstrumentCoverage.scala

```diff
@@ -290,6 +292,15 @@ class InstrumentCoverage extends MacroTransform with IdentityDenotTransformer:
               transformStats(tree.body, tree.symbol)
             )
 
+          case tree: Inlined =>
+            // Ideally, tree.call would provide precise information about the inlined call,
+            // and we would use this information for the coverage report.
+            // But PostTyper simplifies tree.call, so we can't report the actual method that was inlined.
```

Comment:
Isn't tree.call kept as-is by PostTyper? What information are we losing that affects the coverage report exactly?

Reply:
PostTyper replace the `call` tree by a "call trace", see [PostTyper:362](https://github.com/lampepfl/dotty/blob/39b14f6fee7b9823f77fd27d3494cae23cb5899a/compiler/src/dotty/tools/dotc/transform/PostTyper.scala#L362). The [doc of Inlined](https://github.com/lampepfl/dotty/blob/cbdb6103680bf6d3983bcf98aa399e5ccfd5c166/compiler/src/dotty/tools/dotc/ast/Trees.scala#L675) explains:
```
@param  call      Info about the original call that was inlined.
Until PostTyper, this is the full call, afterwards only
a reference to the toplevel class from which the call was inlined.
```
I don't know why it's like this, but the consequence is that for `instrumentCoverage`, `tree.call` only contains the toplevel class, not the original inlined call.

Reply:
Ideally, for:
```scala
assert(x) // inline def in Predef
```
we would report that `assert` was called, but that's not possible for the moment.